google/ggrc-core
# GGRC-5885 Assessments where imported in spite of errors

# Issue description

Import assessments with mappings to objectives raises error but create assessments in database.

# Steps to test the changes

Test case 1:
Steps to reproduce:
1. Create a program
2. Create an audit in scope of the program
3. Create at least 10 objectives in scope of the program (on UI or via import)
4. Download Assessment import template
5. Add 10 Assessments, map 10 Objectives from the program to Assessments (1 Objective to each Assessment). Use 'map:objective versions' column for mapping [see attached csv template]
6. Import Assessment csv file
7. Refresh audit page and open Assessments tab
Actual Result: Import failed due to errors.
Assessments were created on UI w/o snapshots mapping.
NOTE: this is a correct behavior that there was no mapping between Assessment and Objectives, due to original Objectives cannot be mapped to Assessment, only Objective snapshot can be mapped and should exist in audit scope.
Expected Result: Errors that Assessments cannot be imported should be displayed on analysis stage. If errors are shown up, user should download his file, fix the errors and try importing again.

Test case 2
1. Use steps 1-7 from Test case 1
2. Open Audit info tab and click 'Update the audit to the latest version' in 3 dot's menu
3. Ensure that 10 objectives snapshots are displayed in audit scope
4. Export 10 Assessments to Google Sheet
5. Map 10 Objective snapshots to Assessments (1 Objective snapshot to 1 Assessment)
6. Import Assessments
Actual Result: 'Import failed due to unknown error.' is displayed in UI.
Expected Result: Assessments should be imported. Objective snapshot should be mapped to Assessment.

# Solution description

The solution is to validate mappings on analysis import stage.

# Sanity checklist




# PR Review checklist
